ui/pages/Project/components/FamilyTable/IndividualRow.jsx

```diff
@@ -544,6 +548,10 @@ class IndividualRow extends React.PureComponent {
     // only show active or first/ last inactive samples
     loadedSamples = loadedSamples.filter((sample, i) => sample.isActive || i === 0 || i === loadedSamples.length - 1)
 
+    const phenotypePrioritizationTools = individual.phenotypePrioritizationTools.map(
```

Comment:
So constructing arrays for props can cause unwanted updates in react. It would be better to pass `individual.phenotypePrioritizationTools` to the `DataDetails` component and do any transformation/m mapping in the render method

Reply:
I was doing this originally inside of `DataDetails`:
```
    {individual.phenotypePrioritizationTools.map(
      pp => (
        <div key={pp.tool}>
          <Sample loadedSample={{ sampleType: pp.tool.charAt(0).toUpperCase() + pp.tool.slice(1), loadedDate: pp.createdDate }}/>
        </div>
      ),
    )}
```
but was getting lint error `JSX attribute values should not contain objects created in the same scope` so I moved it out into `IndividualRow` render. 
 
which approach is.. less bad?

Reply:
So honestly, I think the best approach is just to return this from the server formatted the way it needs to be formatted for the display, instead of transforming it. IYou would just change the field names you use in the `JSONObject` aggregation

Reply:
yeah that's the right call